core/src/Kokkos_CopyViews.hpp

```diff
@@ -1493,7 +1493,7 @@ void deep_copy
     Kokkos::fence();
   } else {
     Kokkos::fence();
-    Impl::view_copy(typename dst_type::uniform_runtime_nomemspace_type(dst),typename src_type::uniform_runtime_const_nomemspace_type(src));
+    Impl::view_copy(dst, src);
```

Comment:
Does @crtrott approve of this? it seems like it could break the canonicalization that ETI relies on.

Reply:
I believe so, he suggested this as a quick fix to be followed by moving the logic from within the `view_copy` out to before this call to recover that behavior for ETI. 
@crtrott - does this sound accurate?

Reply:
view_copy is not actually part of the ETI thing. That happens inside of view_copy which again uses the uniform types I believe.

Reply:
Yeah just checked. Basically the view_copy function will now potentially exist in more versions than strictly necessary. The actual kernels are still the same.